cosmos_rl/rollout/vllm_rollout/vllm_rollout_worker.py

```diff
@@ -439,11 +507,26 @@ def policy_to_rollout_unicast(self, command: PolicyToRolloutUnicastCommand):
         This is Policy -> Rollout replica. Will only happen between
         a pair of policy and rollout replica.
         """
+        # lazy initialization of the vllm engine.
+        if not self.rollout.is_engine_initialized():
+            is_for_weight_resume = command.dst_replica_name == self.replica_name
+            load_format = "auto" if is_for_weight_resume else "dummy"
+            self.rollout.init_engine(
```

Comment:
Seems only here fo the init engine operation. What about the replicas which don't do the P2R with only R2R, will they do the init engine?